src/js_native_api.h

```diff
@@ -171,6 +171,9 @@ NAPI_EXTERN napi_status napi_get_prototype(napi_env env,
 NAPI_EXTERN napi_status napi_get_property_names(napi_env env,
                                                 napi_value object,
                                                 napi_value* result);
+NAPI_EXTERN napi_status napi_get_own_property_names(napi_env env,
```

Comment:
This needs to go in as experimental and be guarded so only visible when EXPERIMENTAL features are enabled. Should also be documented as experimental in doc above.

Reply:
More details on the rational in https://github.com/nodejs/node/blob/master/doc/guides/adding-new-napi-api.md

and look for `#ifdef NAPI_EXPERIMENTAL` in the file for other examples.